core/src/main/scala/org/apache/spark/api/java/JavaRDDLike.scala

```diff
@@ -436,6 +436,12 @@ trait JavaRDDLike[T, This <: JavaRDDLike[T, This]] extends Serializable {
   def first(): T = rdd.first()
 
   /**
+   * @return true if and only if the RDD contains no elements at all. Note that an RDD
+   *         may be empty even when it has at least 1 partition.
+   */
+  def isEmpty(): Boolean = rdd.isEmpty()
```

Comment:
I've seen code that _accepts_ JavaRDDLike (e.g. as a parameter type), but I can't think of a good reason to extend JavaRDDLike in user code (I couldn't find any non-Spark-internal examples on GitHub code search).  Therefore, I'd be in favor of just adding a MiMa override and updating the documentation to clarify that this interface isn't really designed for end-users.